client/src/sections/courses/learning-section/index.tsx

```diff
@@ -0,0 +1,144 @@
+    { id: 2, progress: 100, firstname: 'John', lastname: 'Doe' },
+    { id: 3, progress: 57, firstname: 'Francis', lastname: 'Delos Santos' },
+    { id: 4, progress: 85, firstname: 'Vali', lastname: 'Ruziboev' },
+    { id: 5, progress: 97, firstname: 'Mel Anthony', lastname: 'Ando' },
+    { id: 6, progress: 75, firstname: 'Zion Keenen', lastname: 'Tavera' },
+    { id: 7, progress: 40, firstname: 'Rene Angelo', lastname: 'Gunayon' },
+    { id: 8, progress: 72, firstname: 'Jason', lastname: 'Chua' },
+    { id: 9, progress: 63, firstname: 'EJ', lastname: 'Potot' },
+    { id: 10, progress: 12, firstname: 'Johny', lastname: 'Shen' },
+    { id: 11, progress: 12, firstname: 'Johny2', lastname: 'Shen' },
+    { id: 12, progress: 63, firstname: 'EJ2', lastname: 'Potot' },
+    { id: 13, progress: 72, firstname: 'Jason2', lastname: 'Chua' },
+    { id: 14, progress: 40, firstname: 'Rene Angelo2', lastname: 'Gunayon' },
+    { id: 15, progress: 75, firstname: 'Zion Keenen2', lastname: 'Tavera' },
+    { id: 16, progress: 97, firstname: 'Mel Anthony2', lastname: 'Ando' },
+    { id: 17, progress: 85, firstname: 'Vali2', lastname: 'Ruziboev' },
+    { id: 18, progress: 57, firstname: 'Francis2', lastname: 'Delos Santos' },
+    { id: 19, progress: 23, firstname: 'Elyric2', lastname: 'Manatad' }
+  ];
+
+  const handleAddLearnersModal = (): void => {
+    setIsModalOpen(true);
+  };
+
+  const handleModalClose = (): void => {
+    setIsModalOpen(false);
+  };
+
+  const handleSeeMore = (): void => {
+    setShowMore(!showMore);
+  };
+
+  const handleSortOptionChange = (value: string): void => {
+    setSelectedSortOption(value);
+  };
+
+  const sortOptions: SortOption[] = [
+    { label: 'A - Z', value: 'A - Z' },
+    { label: 'Z - A', value: 'Z - A' },
+    {
+      label: 'Progress',
+      value: 'progress-high',
+      icon: <ArrowIcon className="transform rotate-90" />
+    },
+    {
+      label: 'Progress',
+      value: 'progress-low',
+      icon: <ArrowIcon className="transform -rotate-90" />
+    }
+  ];
+
+  const sortLearners = (option: string): Learner[] => {
+    switch (option) {
+      case 'A - Z':
+        return learners.sort((a, b) => a.firstname.localeCompare(b.firstname));
+      case 'Z - A':
+        return learners.sort((a, b) => b.firstname.localeCompare(a.firstname));
+      case 'progress-low':
+        return learners.sort((a, b) => a.progress - b.progress);
+      case 'progress-high':
+        return learners.sort((a, b) => b.progress - a.progress);
+      default:
+        return learners;
+    }
+  };
+
+  const sortedLearners = sortLearners(selectedSortOption);
+  const visibleLearners = showMore ? sortedLearners : sortedLearners.slice(0, 9);
+
+  return (
+    <Fragment>
+      <div>
+        <div className="w-full flex items-center justify-between mb-8">
+          <div className="font-semibold text-sm">List of Learners</div>
+          <Button
```

Comment:
you can move this button to the modal component and replace this as the component itself, in that way, we can put all the logics related to modal in one component only, 

you can refer to this 
https://github.com/framgia/sph-lms/pull/98/files#diff-22279cfab9a3f13c060357fad7554a4bda7b2409779e0d64554a5e7eaa924e0d

Reply:
Actually that button is outside of that modal it's just opening and closing it. :bow: